rnahybrid.gap

```diff
@@ -705,6 +707,154 @@ algebra alg_probing implements sig_rnahybrid(alphabet = char, answer = double) {
   }
 }
 
+algebra alg_leftstacklen implements sig_rnahybrid(alphabet = char, answer = int) {
+  int nil(<Subsequence qregion, Subsequence tregion>) {
+    return 0;
+  }
+  int target_left_flank(<Subsequence tregion, Subsequence mloc>, int x) {
+    return x;
+  }
+  int ulb(<Subsequence qloc, Subsequence tbase>, int x) {
+    return x;
+  }
+  int eds(<Subsequence qbase, Subsequence tbase>, int x) {
+    return x;
+  } 
+  int edt(<Subsequence qbase, Subsequence tloc>, int x) {
+    return x;
+  }
+  int edb(<Subsequence qloc, Subsequence tbase>, int x) {
+    return x;
+  }
+  int sr(<Subsequence qbase, Subsequence tbase>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res + 1;
+    }
+    return res;
+  }  
+  int bt(<Subsequence qbase, Subsequence tbase>, <Subsequence qregion, Subsequence tloc>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res * -1;
+    }
+    return res;
+  }
+  int bb(<Subsequence qbase, Subsequence tbase>, <Subsequence qloc, Subsequence tregion>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res * -1;
+    }
+    return res;
+  }
+  int il(<Subsequence qbase, Subsequence tbase>, <Subsequence qregion, Subsequence tregion>, int x) {
+    int res = x;
+    if (res >= 0) {
+      res = res * -1;
+    }
+    return res;
+  }
+  int el(<Subsequence qbase, Subsequence tbase>, <Subsequence qregion, Subsequence tregion>) {
+    return 1;
+  }
+  int complete(int x) {
+    if (x < 0) {
+      return -1 * x;
```

Comment:
It's not clear to me why you change the sign in the end again. Could you explain that to me?

Reply:
In my mind, it would make no sense to have a neative long sequence as a return value, it could also mess with further calculations, no?

Reply:
I intend to count number of a) basepairs (leftstacklen) or b) unpaired bases (mirnabuldgelen). This number should be non-negative in principle. However, I use the sign flip as a hack to stop increasing this number once a candidate is extended with a) a bulge or internal loop or b) by any further element, respectively.

In principle, computation is fine with this, however it is not intuitive to have negative lengths. Since I know that `complete` will always be the root of any candidate (except of nil), I thought it is handy to flip the sign back here.